test/e2e/cli/pkg_unsource_test.go

```diff
@@ -0,0 +1,362 @@
+					Expect(cfgMapUnikernelManifests).To(HaveLen(1))
+					Expect(cfgMapUnikernelManifests[0]).To(Equal("https://manifests.kraftkit.sh/index.yaml"))
+
+					// Check if stdout is empty
+					Expect(stdout.String()).To(BeEmpty())
+				})
+			})
+
+			When("a config file was already presend, and the second unsourced link is duplicate", func() {
+				BeforeEach(func() {
+					unsourceArgs := make([]string, len(cmd.Args))
+					copy(unsourceArgs, cmd.Args)
+
+					for idx, arg := range cmd.Args {
+						if arg == "unsource" {
+							cmd.Args[idx] = "source"
+							break
+						}
+					}
+					cmd.Args = append(cmd.Args, "https://example1.com")
+					cmd.Args = append(cmd.Args, "https://example2.com")
+					cmd.Args = append(cmd.Args, "https://example3.com")
+					err := cmd.Run()
+
+					Expect(err).ToNot(HaveOccurred())
+					Expect(stderr.String()).To(BeEmpty())
+
+					// Recreate the command to reuse
+					stdout = fcmd.NewIOStream()
+					stderr = fcmd.NewIOStream()
+
+					cmd = fcmd.NewKraft(stdout, stderr, cfg.Path())
+					cmd.Args = unsourceArgs
+				})
+
+				It("should remove the first two links and print a warning", func() {
+					cmd.Args = append(cmd.Args, "https://example1.com")
+					cmd.Args = append(cmd.Args, "https://example2.com")
+					cmd.Args = append(cmd.Args, "https://example2.com")
+					err := cmd.Run()
+
+					Expect(err).ToNot(HaveOccurred())
+					Expect(stderr.String()).To(BeEmpty())
+
+					// Read the config file
+					osFile, err := os.Open(cfg.Path())
+					Expect(err).ToNot(HaveOccurred())
+					defer osFile.Close()
+
+					osFileInfo, err := osFile.Stat()
+					Expect(err).ToNot(HaveOccurred())
+
+					// Check if the config file is not empty
+					Expect(osFileInfo.Size()).To(BeNumerically(">", 0))
+
+					// Check if the config file has 600 permissions
+					Expect(osFileInfo.Mode().Perm()).To(Equal(os.FileMode(0o600)))
+
+					// Read file content
+					readBytes, err := os.ReadFile(cfg.Path())
+					Expect(err).ToNot(HaveOccurred())
+
+					// Marshal the yaml file into a map
+					var cfgMap map[string]interface{}
+					err = yaml.Unmarshal([]byte(readBytes), &cfgMap)
+					Expect(err).ToNot(HaveOccurred())
+
+					// Check if the config file contains the default manifests
+					cfgMapUnikernel, ok := cfgMap["unikraft"].(map[string]interface{})
+					Expect(ok).To(BeTrue())
+
+					// Cast the manifests list to an array
+					cfgMapUnikernelManifests, ok := cfgMapUnikernel["manifests"].([]interface{})
+					Expect(ok).To(BeTrue())
+
+					// Check if the default manifest and the third link is still there
+					Expect(cfgMapUnikernelManifests).To(HaveLen(2))
+					Expect(cfgMapUnikernelManifests[0]).To(Equal("https://manifests.kraftkit.sh/index.yaml"))
+					Expect(cfgMapUnikernelManifests[1]).To(Equal("https://example3.com"))
+
+					// Check if stdout has a warning
+					Expect(stdout.String()).To(MatchRegexp(`^{"level":"warning","msg":"manifest not found: https://example2\.com"}\n$`))
```

Comment:
nit: We could as well use `Equal()` here if we are gating the string between `^$`.